lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/expression/AggregateFunctionRule.java

```diff
@@ -27,11 +26,11 @@
 import static com.google.common.base.Verify.verifyNotNull;
 import static java.util.Objects.requireNonNull;
 
-public interface AggregateFunctionRule
+public interface AggregateFunctionRule<Result>
 {
     Pattern<AggregateFunction> getPattern();
 
-    Optional<JdbcExpression> rewrite(AggregateFunction aggregateFunction, Captures captures, RewriteContext context);
+    Optional<Result> rewrite(AggregateFunction aggregateFunction, Captures captures, RewriteContext context);
 
     interface RewriteContext
```

Comment:
RewriteContext for pinot has an extra `getExistingGroupingColumns` method which is used in the count distinct implementation. Can the signature be made generic, changed to something like `<T extends RewriteContext>`? This way other connectors can extended it as needed. Or is there a better way to pass this information?

Reply:
Can you help me understand what is this for? Let's talk about this in https://github.com/trinodb/trino/pull/6069/files#r676189243

Reply:
This was resolved by deciding to do the validation within `applyAggregation` instead of the rewrite rule for Pinot.
Once other non-JDBC connectors start using these classes we'll have a better idea if the RewriteContext needs changes and what those changes should be.